modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/ElasticsearchSearchEngine.java

```diff
@@ -367,7 +399,9 @@ private void _checkNodeVersions() {
 		}
 	}
 
-	private String _createAutoCreateIndexSetting(boolean enable) {
+	private String _createAutoCreateIndexSetting(boolean enable)
+		throws IOException {
+
 		String currentValue = _getAutoCreateIndexSetting();
```

Comment:
Goal #1 in this task says "In the code, check the existence of this setting before even trying to set it. If it was set manually, no warnings would appear in any scenario."

i guess my first concern is if we can actually determine "If it was set manually"... the default is `true`, so if its not true then i guess it was set manually (unless the user manually set it to true, or wanted it to be true)

but lets say we can check it... this seems to be the only place the current value is being checked, and it doesnt seem to have any effect on how the warnings are determined, so does this pr satisfy this goal?

Reply:
I commented in the ticket already that that requirement does not make sense, because even reading the settings can fail because of lacking the permissions.

Reply:
just my opinion but i dont necessarily agree with this goal. i think it can be helpful for an admin to know that we are attempting to update this setting, regardless of what they have/havent done with it themselves @lipusz

Reply:
> just my opinion but i dont necessarily agree with this goal

Which goal specifically? You mean using WARN instead of ERROR or something else? I'd totally agree about using error (see [below](https://github.com/liferay-search/liferay-portal/pull/1514#discussion_r1803703313))